src/cryptography/x509.py

```diff
@@ -249,6 +264,96 @@ def __repr__(self):
                 "path_length={0.path_length})>").format(self)
 
 
+class GeneralName(object):
+    _gn_str_types = [
+        GeneralNameType.rfc822_name,
+        GeneralNameType.dns_name,
+        GeneralNameType.uniform_resource_identifier,
+    ]
+
+    def __init__(self, type, value):
+        if not isinstance(type, GeneralNameType):
+            raise TypeError("type must be a GeneralNameType")
+
+        if type in [
+            GeneralNameType.other_name,
+            GeneralNameType.x400_address,
+            GeneralNameType.edi_party_name
+        ]:
+            raise ValueError("{0} is not a supported type".format(type))
+
+        if (
+            type in self._gn_str_types and not
+            isinstance(value, six.string_types)
```

Comment:
I don't see "idna" anywhere in this PR's changed files, so I'm assuming that while the hunk may be outdated, the sentiment isn't.

You should know that the stdlib "idna" encoding is somewhat out of date, and according to @hynek you should prefer [the separately-distributed `idna` package](https://pypi.python.org/pypi/idna).